### PR TITLE
Adding specific items into troubleshooting guide

### PR DESCRIPTION
**Pull request by Leaf-Lin**
I think having the itemised list separated by health component (Data, Cluster Coordination, Backup, Tasks etc) will help user find the pages without checking the sidebar. 

It's likely that the syntax I added in the ASCIIDOC is not quite right for what I want to do similar to the itemised list in https://www.elastic.co/guide/en/elasticsearch/reference/master/recipes.html

Please help backport this to 8.3.0.

### CODE REVIEW

File: docs/reference/troubleshooting.asciidoc [hunk 1]
```diff
@@ -56,6 +77,8 @@ include::troubleshooting/data/start-ilm.asciidoc[]
 
 include::troubleshooting/data/start-slm.asciidoc[]
 
+include::troubleshooting/data/restore-from-snapshot.asciidoc[]
```
Comment (abdonpijpelink) [inline docs/reference/troubleshooting.asciidoc:80]: @Leaf-Lin This page does not exist. Is this a new page you want to add, or did you mean to link to https://www.elastic.co/guide/en/elasticsearch/reference/master/snapshots-restore-snapshot.html ?

Comment (Leaf-Lin) [reply docs/reference/troubleshooting.asciidoc:80]: This refers to https://www.elastic.co/guide/en/elasticsearch/reference/master/restore-from-snapshot.html just published via https://github.com/elastic/elasticsearch/pull/87692

**Comment by abdonpijpelink** [comment]
@Leaf-Lin I've organized the links into groups. What do you think? https://elasticsearch_88105.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/troubleshooting.html

**Comment by Leaf-Lin** [comment]
LGTM, thanks!

**Review by abdonpijpelink — APPROVED** [review]

**Pull request merged**